src/librustc/dep_graph/dep_node.rs

```diff
@@ -56,7 +58,7 @@ pub enum DepNode<D: Clone + Debug> {
 
     /// Represents some artifact that we save to disk. Note that these
     /// do not have a def-id as part of their identifier.
-    WorkProduct(Arc<WorkProductId>),
+    WorkProduct(WorkProductId),
```

Comment:
Yes, `DepNode` will be `Copy`.
A `Fingerprint` is a hash value with a "virtual uniqueness" property, i.e. there will be collisions (like with any hash) but the probability is so low that it doesn't matter. Because of this property, a `Fingerprint` can be used as an identifier. In terms of `rustc` I now use the type `Fingerprint` for anything that fulfills this property but sometimes I like to introduce newtypes for it, for example `DefPathHash`.